airflow/cli/api/client.py

```diff
@@ -0,0 +1,244 @@
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import json
+import os
+import sys
+from typing import TYPE_CHECKING, Any
+
+import httpx
+import keyring
+import rich
+import structlog
+from platformdirs import user_config_path
+from uuid6 import uuid7
+
+from airflow.cli.api.operations import (
+    AssetsOperations,
+    BackfillsOperations,
+    ConfigOperations,
+    ConnectionsOperations,
+    DagOperations,
+    DagRunOperations,
+    JobsOperations,
+    PoolsOperations,
+    ProvidersOperations,
+    ServerResponseError,
+    VariablesOperations,
+    VersionOperations,
+)
+from airflow.utils.types import DagRunType
+from airflow.version import version
+
+if TYPE_CHECKING:
+    # # methodtools doesn't have typestubs, so give a stub
+    def lru_cache(maxsize: int | None = 128):
+        def wrapper(f):
+            return f
+
+        return wrapper
+else:
+    from methodtools import lru_cache
+
+log = structlog.get_logger(logger_name=__name__)
+
+__all__ = [
+    "Client",
+    "Credentials",
+]
+
+
+def add_correlation_id(request: httpx.Request):
+    request.headers["correlation-id"] = str(uuid7())
+
+
+def get_json_error(response: httpx.Response):
+    """Raise a ServerResponseError if we can extract error info from the error."""
+    err = ServerResponseError.from_response(response)
+    if err:
+        log.warning("Server error ", extra=dict(err.response.json()))
+        raise err
+
+
+def raise_on_4xx_5xx(response: httpx.Response):
+    return get_json_error(response) or response.raise_for_status()
+
+
+def noop_handler(request: httpx.Request) -> httpx.Response:
+    path = request.url.path
+    log.debug("Dry-run request", method=request.method, path=path)
+    # TODO change for test
+    if path.startswith("/task-instances/") and path.endswith("/run"):
+        # Return a fake context
+        return httpx.Response(
+            200,
+            json={
+                "dag_run": {
+                    "dag_id": "test_dag",
+                    "run_id": "test_run",
+                    "logical_date": "2021-01-01T00:00:00Z",
+                    "start_date": "2021-01-01T00:00:00Z",
+                    "run_type": DagRunType.MANUAL,
+                },
+            },
+        )
+    return httpx.Response(200, json={"text": "Hello, world!"})
```

Comment:
This handler doesn't make any sense for the CLI

Reply:
Removed